SCClassLibrary/Common/Core/Error.sc

```diff
@@ -147,42 +147,45 @@ ShouldNotImplementError : MethodError {
 }
 
 DoesNotUnderstandError : MethodError {
-	var <>selector, <>args, <suggestedCorrection, suggestion;
+	var <>selector, <>args, <suggestedCorrection, suggestion = "";
 	*new { arg receiver, selector, args;
 		^super.new(nil, receiver).selector_(selector).args_(args).init
 	}
+
 	init {
 		var methods, methodNames, editDistances, minIndex, lowerCaseSelector;
-		methods = receiver.class.superclasses.add(receiver.class).collect(_.methods).reduce('++');
-		methodNames = methods.collect { |x| x.name.asString.toLower };
-		// we compare lower-case versions to prioritize capitalization mistakes
-		lowerCaseSelector = selector.asString.toLower;
-		editDistances = methodNames.collect(_.editDistance(lowerCaseSelector));
-		minIndex = editDistances.minIndex;
-		// Edit distance of 3 chosen arbitrarily; also, filter out completely dissimilar matches
-		// to avoid unhelpful suggestions.
-		if(editDistances[minIndex] <= 3 and: { methodNames[minIndex].similarity(lowerCaseSelector) > 0 }) {
-			suggestedCorrection = methods[minIndex];
-			suggestion = " Did you mean '%'?".format(suggestedCorrection.name);
-		} {
-			suggestedCorrection = nil;
-			suggestion = "";
+
+		if(receiver.notNil) {
+			methods = receiver.class.superclasses.add(receiver.class).collect(_.methods).reduce('++');
+			methodNames = methods.collect { |x| x.name.asString.toLower };
+			// we compare lower-case versions to prioritize capitalization mistakes
+			lowerCaseSelector = selector.asString.toLower;
+			editDistances = methodNames.collect(_.editDistance(lowerCaseSelector));
+			minIndex = editDistances.minIndex;
+			// Edit distance of 3 chosen arbitrarily; also, filter out completely dissimilar matches
+			// to avoid unhelpful suggestions.
+			if(editDistances[minIndex] <= 3 and: { methodNames[minIndex].similarity(lowerCaseSelector) > 0 }) {
+				suggestedCorrection = methods[minIndex];
+				suggestion = "\nPerhaps you misspelled '%', or meant to call '%' on another receiver?"
+				.format(suggestedCorrection.name, selector);
+			}
 		}
 	}
 	errorString {
 		^"ERROR: Message '" ++ selector ++ "' not understood." ++ suggestion
 	}
 	reportError {
+		"\n".post;
 		this.errorString.postln;
-		"RECEIVER:\n".post;
+		"\nRECEIVER:\n".post;
 		receiver.dump;
 		"ARGS:\n".post;
 		args.dumpAll;
 		this.errorPathString.post;
 		if(protectedBacktrace.notNil, { this.postProtectedBacktrace });
 		this.dumpBackTrace;
 		// this.adviceLink.postln;
-		"^^ The preceding error dump is for %\nRECEIVER: %\n\n\n".postf(this.errorString, receiver);
+		"\n^^ %\nRECEIVER: %\n\n\n".postf(this.errorString, receiver);
```

Comment:
please keep this as it was, the sections of the printout are very clearly labelled and extra vertical whitespace just leads to more scrolling.

Reply:
These are only 3 lines out of 40. Maybe give it a second look tomorrow, or we ask others what they think? These things are sometimes a matter of taste, in the end I'm fine with either of course. To me, it is an improvement.

Also I'd really like to get rid of the long "The preceding error dump is for" – it makes it harder to read, and the "^^" are clear indications upward.

Reply:
i don't want to solve formatting issues by adding extra blank lines in the middle of a printout; it creates a bad precedent, which i explained before [here](https://github.com/supercollider/supercollider/pull/4476#discussion_r419421204). i would be much more interested in removing unnecessary information, or changing the way the information is presented, but that would happen outside this PR.

Reply:
OK, I'm not sure I understand your problem, but fine with me! Let's keep this in mind though.